src/main/java/org/jboss/pull/player/PullPlayer.java

```diff
@@ -76,21 +77,38 @@ private void processPulls(PersistentList whiteList, PersistentList adminList, Li
             // Add the pull to the label processor
             labelProcessor.add(pull);
 
-            String job = Jobs.getCompletedJob(sha1);
-
             boolean help = false;
             boolean retrigger = false;
             boolean retriggerFailed = false;
             Instant retriggerDate = null;
             boolean whitelistNotify = true;
             String commentId = "";
 
-            String commentsUrl = pull.get("comments_url").asString(); //get url for comments
-            boolean mergeable = pull.get("mergeable").asString().equals("true");
+            ModelNode prDetails = gitHubApi.getPullRequestDetails(pullNumber);
 
+            // not all pr information is available on the pr list, and in order for a test merge to be created
+            // the pr details have to be either fetched or looked at in the browser. If mergeable is false or null
+            // then the pr is skipped and rechecked on the next run
+            // see: https://developer.github.com/v3/git/#checking-mergeability-of-pull-requests
+            boolean mergeable = prDetails.get("mergeable").asString("false").equals("true");
+            if (!mergeable) {
+                System.out.println("PR is not marked as mergeable yet, skipping.");
+                continue;
+            }
+            // this is the test commit created that is a merge onto the branch the pr has been opened against.
+            // see: https://developer.github.com/v3/pulls/#response-1
+            String mergeCommitSha = prDetails.get("merge_commit_sha").asString();
+            if (mergeCommitSha == null) {
+                System.out.println("No vaild merge_commit_sha found on PR, skipping.");
+                gitHubApi.postComment(pullNumber, "There was no valid merge_commit_sha found on this PR though mergeable is set to true, it will be retried.");
+                continue;
+            }
```

Comment:
I'm not sure all the processing below should be skipped in the mergeable = false (as opposed to null) case. IOW if the PR has a merge conflict the pull player won't respond to anything, but in some cases perhaps it should

Reply:
Ah, yeah good point, I'll make this more selective so commands still work, thanks!

Reply:
But probably not worth the complexity of dealing with that for a fairly understandable situation -- GH tells you if you have a conflict so it's not hard to understand why the pull player might not be doing something.

Sp LGTM.

Reply:
Its a pretty easy fix to just delay the check until after the command loop, so i just pushed that.